paddle/fluid/operators/dgc_clip_by_norm_op.cc

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
+
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <string>
+
+#include "paddle/fluid/operators/dgc_clip_by_norm_op.h"
+
+namespace paddle {
+namespace operators {
+
+class DGCClipByNormOp : public ClipByNormOp {
+ public:
+  using ClipByNormOp::ClipByNormOp;
+
+ protected:
+  void InferShape(framework::InferShapeContext* ctx) const override {
+    PADDLE_ENFORCE(ctx->HasInput("current_step"),
+                   "current_step should be set.");
+
+    return ClipByNormOp::InferShape(ctx);
+  }
+
+  framework::OpKernelType GetKernelTypeForVar(
+      const std::string& var_name, const framework::Tensor& tensor,
+      const framework::OpKernelType& expected_kernel_type) const override {
+    if (var_name == "current_step") {
+      VLOG(10) << "var_name:" << var_name << " need not to transform";
+      return expected_kernel_type;
+    }
+
+    return framework::OperatorWithKernel::GetKernelTypeForVar(
+        var_name, tensor, expected_kernel_type);
+  }
+};
+
+class DGCClipByNormOpMaker : public ClipByNormOpMaker {
+ public:
+  void Make() override {
+    AddInput("current_step", "(Tensor) Current step.");
+    AddAttr<float>("rampup_begin_step",
+                   "(float, -1.0)"
+                   "The period when begin k_select.")
+        .SetDefault(-1.0);
+
+    return ClipByNormOpMaker::Make();
```

Comment:
The doc of `DGCClipByNormOp` and `ClipByNormOpMaker` should be different.

Reply:
It's different now.And the child calls parents' function.